h5_replace_gps.py

```diff
@@ -181,60 +160,175 @@ def substituteXMLval(name, newval, xml):
                               gpscluster)
-                gpstimestamp = re.search('[0-9]{6}', m.group()).group()
-                hdftimes.append(get_time(gpstimestamp,
-                                         dataset.attrs["PCSavetimestamp"],
-                                         tzoffset))
-
+                if m == None:  
+                    # if there is no GPS timestamp - use the one from the EPU computer
+                    hdfgpstimes.append(np.nan)
+                else:
+                    # if there is a GPS timestamp use that instead
+                    gpstimestamp = re.search('[0-9]{6}\.?\d*', m.group()).group()
+                    hdfgpstimes.append(get_time(gpstimestamp,
+                                         timestamp, 0))  # GPS time IS in UTC so tzoffset must be 0
             except (ValueError, AttributeError):
-                hdftimes.append(np.nan)
-
-# Interpolate the GPS positions
-hdfseconds = np.array([dateseconds(a) for a in hdftimes])
-gpsseconds = np.array([dateseconds(a) for a in gpstimes])
-
-def sortby(a, b):
-    return [a_ for (b_, a_) in sorted(zip(b,a))]
-
-gpslons = sortby(gpslons, gpsseconds)
-gpslats = sortby(gpslats, gpsseconds)
-gpseles = sortby(gpseles, gpsseconds)
-gpsseconds.sort()
-
-interp_lons = np.interp(hdfseconds, gpsseconds, gpslons)
-interp_lats = np.interp(hdfseconds, gpsseconds, gpslats)
-interp_eles = np.interp(hdfseconds, gpsseconds, gpseles)
-
-# Create a mask indicating where to use the interpolants
+                hdfgpstimes.append(np.nan)
+
+# convert all datetimes to seconds
+hdfgpsseconds = np.array([dateseconds(a) for a in hdfgpstimes])  # represents the timestamps from IPR GPS
+hdfpcseconds = np.array([dateseconds(a) for a in hdfpctimes])  # represents the timestamps from  EPU computer
+gpsseconds = np.array([dateseconds(a) for a in gps.gpstimes])  # represents the timestamps from 'better' GPS file provided
+
+#DM compared hdfgpstimes and hdfpctimes for some files: 
+# What is the offset?  Can you use pc time instead of gps time when it is missing? 
+# I would say no.  The difference between these is variable and there will be an offset
+# between gps and pc time unless the pc was just set to UTC... 
+#np.nanmean(hdfgpsseconds-hdfpcseconds)
+#np.nanmax(hdfgpsseconds-hdfpcseconds)
+#np.nanmin(hdfgpsseconds-hdfpcseconds)
+#np.nanmedian(hdfgpsseconds-hdfpcseconds)
+#np.nanstd(hdfgpsseconds-hdfpcseconds)
+# You CAN use pctimes instead of the gps times but this is not recommended (see timesource) 
+# This may be improved if the pc times are lagged accordingly
+
+if args.timesource == 'iprgps':
+    hdfseconds = hdfgpsseconds
+elif args.timesource == 'pcgps':    
+    hdfseconds = hdfpcseconds
+elif args.timesource == 'both':
+    hdfseconds = hdfgpsseconds
+    # if the hdfgps data is nan, then use the other source
+    hdfseconds[np.isnan(hdfseconds) == True] = hdfpcseconds[np.isnan(hdfgpsseconds) == True]
```

Comment:
```suggestion
    hdfseconds[np.isnan(hdfgpsseconds)] = hdfpcseconds[np.isnan(hdfgpsseconds)]
```
A bit easier to read when it's obvious that we're using the same boolean mask on the LHS and RHS.
(Also, `np.isnan` returns an array of booleans, and we don't need to ask whether `True == True` or `False == True`).